widget/entry.go

```diff
@@ -908,6 +908,53 @@ func (e *Entry) Unbind() {
 	e.binder.Unbind()
 }
 
+// InsertAtCursor inserts the given text into the Entry at the current cursor position.
+// If text is currently selected, the selected text will be replaced
+// with the inserted content.
+//
+// Since: 2.6
+func (e *Entry) InsertAtCursor(text string) {
+	if text == "" {
+		changed := e.selecting && e.eraseSelection()
+
+		if changed {
+			e.Refresh()
+		}
+
+		return // Nothing to paste into the text content.
+	}
+
+	if !e.MultiLine {
```

Comment:
I think this code is specific to clipboard so maybe it should be outside of the new insert API?

If you think it does belong here please document it in the API doc.

Reply:
@matwachich Do you have an opinion on whether the InsertAtCursor API should also reformat multiline text when inserting into a single-line entry (like paste does)? I feel like it maybe shouldn't since it'd be the developer who controls what is being inserted, vs a user action like paste.